client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you.tsx

```diff
@@ -89,24 +91,30 @@ const MarketplaceThankYou = ( { productSlug }: { productSlug: string } ) => {
 		! new URLSearchParams( document.location.search ).has( 'hide-progress-bar' )
 	);
 
-	const areAllPluginsOnSite =
-		!! pluginsOnSite.length && pluginsOnSite.every( ( pluginOnSite ) => !! pluginOnSite );
+	// Retrieve theme information
+	const wpComThemes = useSelector( ( state ) => getThemes( state, 'wpcom', productSlugs ) );
+
+	const isAllDataFetched =
+		!! pluginsOnSite.length &&
+		pluginsOnSite.every( ( pluginOnSite, index ) => !! pluginOnSite || !! wpComThemes[ index ] );
```

Comment:
Do `pluginsOnSite` also return themes? I am not sure why you have added `!! wpComThemes[ index ]` here

Reply:
Would moving the purchase items (maybe the whole cart before it gets truncated?) to state be helpful?

Reply:
We are currently working with this page being stateless, this way it could be refreshed/shared and the user would still be able to see the same results. 

This page is also used for not purchased items (ex: installed plugins) and we handle it the same way as the purchased ones, if we start to work with the purchase objects it would need to be handled as an edge case.

Reply:
> refreshed

We could persist the data in the browser.

> shared

I am not sure we have such a business case?

> This page is also used for not purchased items (ex: installed plugins) and we handle it the same way as the purchased ones

That's a fair point. 

I wonder how we can implement the changes in p9Jlb4-6qr-p2 in the future if we have all products in the URL.

Currently, this means:
- parsing the product slugs
- querying to find out their type
- querying to find out more data based on their type

Am I correct?

Reply:
> We could persist the data in the browser.

Good point, I haven't considered it.

> I am not sure we have such a business case?

I can't think of one, I just pointed it out.

> Am I correct?

I think its a bit worse request-wise, but not time-wise.
We actually send all requests in parallel and  check what results we get.

Using the example above of 3 slugs, we would send:
* 3 requests to a wporg search
* 3 requests to a wpcom search
* 3 requests for themes

Once we get all of them from at least one source, we show the page. 

Considering we'll have even more product types in the future, I agree its a good idea to try to find a better alternative to retrieve it.

Reply:
What do you think about using a query string like `?themes=slug1,slug2&plugins=slug1,slug2&plan=slug`?

Reply:
+1 to this, I think it'd be great to be able to reduce the number of queries by indicating the product type in the URL

Reply:
> What do you think about using a query string like `?themes=slug1,slug2&plugins=slug1,slug2&plan=slug`?

I think we can work with it. I can't see another way to better define the type from the URL. 

---

That said, I think this PR can be merged for retrieving the theme data and I'll create a new PR for updating how we receive the info from the URL.

Reply:
Ok, let's iterate on it.